### PR TITLE
Should address all the comments we discussed in the PR

### PR DESCRIPTION
**Pull request by ElDeveloper**

### CODE REVIEW

File: q2_qemistree/_itol_metadata.py [hunk 1]
```diff
@@ -40,23 +49,23 @@ def classyfire_to_colors(classified_feature_data: pd.DataFrame,
 def get_itol_visualization(classified_feature_data: str,
                            classyfire_level: str = 'class',
                            color_file_path: str = './itol_colors.txt',
-                           label_file_path: str = './itol_labels.txt')
-                           color_palette: str = 'bright'):
+                           label_file_path: str = './itol_labels.txt',
+                           color_palette: str = 'husl'):
```
Comment (anupriyatripathi) [inline q2_qemistree/_itol_metadata.py:54]: looks great!

File: q2_qemistree/_prune_hierarchy.py [hunk 2]
```diff
@@ -10,64 +10,62 @@
...
+        this column will be removed from the phylogeny. The choices are one of:
+        `'kingdom'`, `'superclass'`, `'class'`, `'subclass'`, `'direct_parent'`
+        and `'smiles'`.
 
     Raises
     ------
     ValueError
-        If user-specified `classyfire_level` is not in `feature_data.columns`
-        If `smiles` not in `feature_data.columns` when `prune_type` = 'smiles'
+        If th
```
Comment (anupriyatripathi) [inline q2_qemistree/_prune_hierarchy.py]: Is this missing some text?

File: q2_qemistree/_prune_hierarchy.py [hunk 3]
```diff
@@ -10,64 +10,62 @@
...
+    if column not in feature_data.columns:
+        raise ValueError("The feature data does not contain the column '%s'" %
+                         column)
+
+    if column not in known:
+        raise ValueError('Pruning cannot be applied on column "%s". The only '
+                         'options are: %s. If your feature data does not '
+                         'include this information, consider using '
+                         'get_classyfire_taxonomy.' % (column,
+                                                       ', '.join(known)))
```
Comment (anupriyatripathi) [inline q2_qemistree/_prune_hierarchy.py:54]: This should be very useful -- thanks for adding!

File: q2_qemistree/_prune_hierarchy.py [hunk 4]
```diff
@@ -10,64 +10,62 @@
...
+
+    failed_values = {'unclassified', 'unexpected server response',
+                     'SMILE parse error'}
+
+    # remove all NA values or missing values
+    feature_data = feature_data[~(feature_data[column].isin(failed_values) |
+                                  feature_data[column].isna())]
+
```
Comment (anupriyatripathi) [inline q2_qemistree/_prune_hierarchy.py:62]: 👍

File: q2_qemistree/tests/test_classyfire.py [hunk 5]
```diff
@@ -44,6 +44,7 @@ def test_nan_smiles(self):
             get_classyfire_taxonomy(self.nan_smiles)
 
     def test_malformed_smiles(self):
+        # TODO: assertwarnsregex
```
Comment (anupriyatripathi) [inline q2_qemistree/tests/test_classyfire.py]: It should be fine for now - thanks for adding the TODO.

File: q2_qemistree/tests/test_prune_hierarchy.py [hunk 6]
```diff
@@ -16,62 +16,75 @@
...
                                                 'unclassified'],
                                           columns=['class'])
-        self.one_annotation = pd.DataFrame(index=['A', 'B', 'D'],
-                                           data=['foo', 'unclassified',
-                                                 'unclassified'],
-                                           columns=['class'])
+        self.invalid_column = pd.DataFrame(index=['A', 'B', 'D'],
+                                           data=[['foo', ':)'],
+                                                 ['unclassified', ':)'],
+                                                 ['unclassified', ':D']],
```
Comment (anupriyatripathi) [inline q2_qemistree/tests/test_prune_hierarchy.py:29]: :D

File: q2_qemistree/_itol_metadata.py [hunk 7]
```diff
@@ -19,6 +22,12 @@ def classyfire_to_colors(classified_feature_data: pd.DataFrame,
     annotations = classified_feature_data[classyfire_level].unique()
     colors = sns.color_palette(color_palette,
                                n_colors=len(annotations)).as_hex()
+
+    # give a heads up to the user
+    if set(colors) < len(annotations):
+        warnings.warn(UserWarning, "The mapping between colors and annotations"
+                      " is not unique, some colors have been repeated")
```
Comment (anupriyatripathi) [inline q2_qemistree/_itol_metadata.py]: I think this should be:
```suggestion
warnings.warn("The mapping between colors and annotations"
              " is not unique, some colors have been repeated", UserWarning)
```

**Review by anupriyatripathi — CHANGES_REQUESTED** [review]
Hi Yoshiki, the fixes look really good. I just have a couple of minor edit requests which were probably typos after which it should be ready to merge! I can update the README if you prefer. Thanks!

File: q2_qemistree/_prune_hierarchy.py (hunk 2 above)
Comment (ElDeveloper) [reply q2_qemistree/_prune_hierarchy.py]: Fixed, thanks for catching this.

File: q2_qemistree/tests/test_classyfire.py (hunk 5 above)
Comment (ElDeveloper) [reply q2_qemistree/tests/test_classyfire.py]: Thanks for catching this, I just added a test

**Review by anupriyatripathi — APPROVED** [review]
Looks great - thanks!

**Pull request merged**